recaf-ui/src/main/java/me/coley/recaf/ui/pane/DecompilePane.java

```diff
@@ -123,6 +126,22 @@ private Node createButtonBar() {
 		Label decompilersLabel = new Label("Decompiler: ");
 		box.getChildren().add(decompilersLabel);
 		box.getChildren().add(decompilerCombo);
+
+		if (commonClassInfo != null &&
+				Configs.display().showUnreferencedClassWarning &&
+				commonClassInfo.getSourceType() != ClassSourceType.INTERNAL_LIBRARY && // don't show warning for internal libraries
```

Comment:
This will allow the warning to show in phantom classes, which is another reason why the phantom enum value can be merged into `INTERNAL_LIBRARY` tbh.